step-generation/src/utils/pythonFormat.ts

```diff
@@ -0,0 +1,65 @@
+/** Utility functions for Python code generation. */
+
+export const INDENT = '    '
+
+/** Indent each of the lines in `text`. */
+export function indentLines(text: string): string {
+  return text
+    .split('\n')
+    .map(line => (line ? INDENT + line : line))
+    .join('\n')
+}
+
+/** Render an arbitrary JavaScript value to Python. */
+export function formatPyValue(value: any): string {
+  switch (typeof value) {
+    case 'undefined':
+      return 'None'
+    case 'boolean':
+      return value ? 'True' : 'False'
+    case 'number':
+      // `float("Infinity")` and `float("NaN")` is how you write those values in Python
+      return Number.isFinite(value) ? `${value}` : `float("${value}")`
+    case 'string':
+      return formatPyStr(value)
+    case 'object':
+      if (value === null) {
+        return 'None'
+      } else if (Array.isArray(value)) {
+        return formatPyList(value)
+      } else {
+        return formatPyDict(value)
```

Comment:
maybe you can type this as `Record<string, any>` to make typescript happy

```suggestion
        return formatPyDict(value as Record<string, any>)
```